lib/Sema/LookupVisibleDecls.cpp

```diff
@@ -219,9 +228,17 @@ static void lookupTypeMembers(Type BaseType, Type LookupType,
   SmallVector<ValueDecl*, 2> FoundDecls;
 
   for (Decl *Member : D->getMembers()) {
-    if (auto *VD = dyn_cast<ValueDecl>(Member))
-      if (isDeclVisibleInLookupMode(VD, LS, CurrDC, TypeResolver))
-        FoundDecls.push_back(VD);
+    auto *VD = dyn_cast<ValueDecl>(Member);
```

Comment:
There's code duplication between here and doGlobalExtensionLookup(). While it existed before your change, your change makes it worse, so please clean it up

Reply:
Factored out to `collectVisibleMemberDecls()`